src/js/components/AppListFilterComponent.jsx

```diff
@@ -17,24 +20,78 @@ var AppListFilterComponent = React.createClass({
     };
   },
 
-  updateFilterText: function (event) {
+  componentDidMount: function () {
+    this.updateFilterText();
+  },
+
+  componentWillReceiveProps: function () {
+    this.updateFilterText();
+  },
+
+  setQueryParam: function (filterText) {
+    var router = this.context.router;
+
+    var queryParams = router.getCurrentQuery();
+
+    if (filterText != null && filterText !== "") {
+      Object.assign(queryParams, {
+        filterText: encodeURIComponent(filterText)
+      });
+    } else {
+      delete queryParams.filterText;
+    }
+
+    router.transitionTo(router.getCurrentPathname(), {}, queryParams);
+  },
+
+  updateFilterText: function () {
+    var router = this.context.router;
+    var filterText = router.getCurrentQuery().filterText;
```

Comment:
Makes it slower, because queryParams is used only once here, but goes in line with our chat above.. ;) Changed it.